core/spec/models/spree/order_shipping_method_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe Spree::Order do
+  let(:order) { create(:order_with_line_items) }
+  let(:quadcopter) { create(:shipping_method, name: "Quads'r'us", code: "QUADSRUS", cost: 200) }
+
+  it "preserves selected shipping method when update_cart and next! called" do
+    #choose more expensive quadcopter shipping method
+    original_shipment = order.shipments.first
+    quadcopter_rate = original_shipment.add_shipping_method(quadcopter)
+    original_shipment.selected_shipping_rate_id = quadcopter_rate.id
```

Comment:
You'd think so, but no. `Shipment.selected_shipping_rate_id` does a `save!` itself. I'm not a fan of that, but it's a problem for another day.